src/Pos/Launcher/Scenario.hs

```diff
@@ -97,11 +95,10 @@ waitForPeers = discoverPeers >>= \case
 initSemaphore :: (WorkMode ssc m) => m ()
 initSemaphore = do
     semaphore <- ncBlkSemaphore <$> getNodeContext
-    unlessM
-        (liftIO $ isEmptyMVar semaphore)
-        (logError "ncBlkSemaphore is not empty at the very beginning")
+    whenNothingM_ (tryReadMVar semaphore) $
```

Comment:
Idiomatically, `Bool` is worse than `Maybe a`. Using `try*` functions you can do the same things. And you actually don't really often need `isEmptyMVar`. Also this function has next documentation:

```haskell
-- |Check whether a given 'MVar' is empty.
--
-- Notice that the boolean value returned  is just a snapshot of
-- the state of the MVar. By the time you get to react on its result,
-- the MVar may have been filled (or emptied) - so be extremely
-- careful when using this operation.   Use 'tryTakeMVar' instead if possible.
```
So I don't want someone accidentally use this function because it's not so good.